lib/src/misc/socketio.dart

```diff
@@ -51,24 +52,14 @@ class SocketIOClient extends RenovationController {
 
     config.logger.i('LTS-Renovation-Core-Dart Connecting socket on $url$path');
 
-    if (!_socket!.hasListeners('connect')) {
-      _socket!.on(
-          'connect',
-          (dynamic data) =>
-              config.logger.i('Connected socket successfully on $url$path'));
-    }
-    if (!_socket!.hasListeners('connect_error')) {
-      _socket!.on(
-          'connect_error',
-          (dynamic connectStatus) =>
-              config.logger.e('Connected socket unsuccessful on $url$path'));
-    }
-    if (!_socket!.hasListeners('connect_timeout')) {
-      _socket!.on(
-          'connect_timeout',
-          (dynamic connectStatus) => config.logger
-              .e(<dynamic>['Timeout while connecting', connectStatus]));
-    }
+    _socket!.onConnect((dynamic data) =>
```

Comment:
@nabilshuja I think we still need to guard against duplicating event handlers by using the `hasListeners`

Reply:
The problem with why I removed the `hasListeners` is because internally, for example, it seems to bind an event handler on `connect` itself. However, checking the rest of them (besides `connect`) seem fine now.  

So, do you think instead we should have a field that states whether or not these events handler have been binded instead of relying on `hasListeners`.